### DISCUSSION TITLE
A question about EDA_Phasic

### DISCUSSION THREAD

**Discussion by RyotaTKN** [neuropsychology/NeuroKit#546] [Q&A] [answered]
Dear All

I have one simple question about EDA_Phasic.

I got time-course EDA_Phasic data by using python code as follows.

```python
eda_signals, info = nk.eda_process(df['SC - GSR100C'], sampling_rate=250) # Process the EDA data
eda_phasic = eda_signals.loc[:,['EDA_Phasic']]
```

In this case, what the 'EDA_Phasic' data mean? Is that "Non-specific SCR" or "Specific SCR"?

Thank you in advance!

Sincerely,
Ryota

**Comment by Tam-Pham** [comment, marked as answer]
Hi @RyotaTKN 

Apologize for the late response. Here, EDA phasic refers to the phasic components. In order to identify the SCR features, [nk.eda_peaks()](https://neurokit2.readthedocs.io/en/latest/functions.html#neurokit2.eda.eda_peaks) function can be applied on the `eda_phasic` component. 

The `nk.eda_peaks()` will return all SCRs, both specific and non-specific. For specific SCR or event-related SCR, you can look at the documentation of our [nk.eda_eventrelated()](https://neurokit2.readthedocs.io/en/latest/functions.html#neurokit2.eda.eda_eventrelated) function for more info. Hope that helps!

**Comment by RyotaTKN** [comment]
Thanks for your reply! I understood that the specific-SCR can be obtained by using nk.eda_eventrelated().

Actually, I'd like to get the mean and the time-series change of the SCR while watching 2-minute video (based on the Zickfeld et al., 2019, Psychophysiology).

Hence, I got the EDA Phasic component by using the Python code above.

In this case, is the EDA Phasic data specific? or non-specific? (or is this question itself meaningless?) 
I'd like to get the "phasic change" of the EDA response, not "frequency" of it.

I'm sorry I didn't make it clear enough.
Thank you in advance!

Sincerely,
Ryota

**Reply by Tam-Pham** [reply]
I took a look at [Zickfeld et al., 2019](https://onlinelibrary.wiley.com/doi/pdfdirect/10.1111/psyp.13662) and I think I have a clearer idea of what you are looking for now :)

> In this case, is the EDA Phasic data specific? or non-specific? (or is this question itself meaningless?)

Whether SCRs are considered as specific or non-specific depends on whether it occurs in tandem with an event of interest. However, in your case where the "event of interest" lasts for 2 minutes, the specificity of SCR is not that relevant (since there is only 1 event and presumably many SCRs in 2 mins, they are all considered to be non-specific SCRs). And that's why Zickfeld chose to calculate the frequency of SCRs instead. Neurokit does have a `nk.eda_intervalrelated()` function that returns the total number of SCRs occurred in a period of time. 

> Actually, I'd like to get the mean and the time-series change of the SCR while watching 2-minute video (based on the Zickfeld et al., 2019, Psychophysiology).
>I'd like to get the "phasic change" of the EDA response, not "frequency" of it.

 I might need your help to clarify what output you are hoping to extract from your EDA signal. As SCRs are the peaks that occur throughout the recordings, do you mean you want to get the mean of the phasic component (which is continuous)?

**Reply by RyotaTKN** [reply]
Thank you for your kind help!

>Do you mean you want to get the mean of the phasic component (which is continuous)?

Yes! Exactly.  I would like to get the mean and continuous data of the phasic component.

**Reply by Tam-Pham** [reply]
If it's just the mean of the phasic component, I imagine you can just `np.mean` the continuous phasic component that was returned in  your code

```python
eda_signals, info = nk.eda_process(df['SC - GSR100C'], sampling_rate=250) # Process the EDA data
eda_phasic = eda_signals.loc[:,['EDA_Phasic']]
```
Or is there any other quantification of this component that you are hoping to have?

**Reply by RyotaTKN** [reply]
Thank you for your reply.

I would like to know what the continuous phasic component while watching 2-minute emotional video means.
In other word, what should we call the measurement when writing an article?

According to the definitions as following, is that "non-specific SCR"? 
This table is from: https://www.biopac.com/wp-content/uploads/EDA-SCR-Analysis.pdf

Please let me know if there is anything else you would like to know about my experimental data or purpose.

![image](https://user-images.githubusercontent.com/58549690/134323171-8b478d46-9383-48de-9c31-f11efdd166e3.png)
